api_fhir_r4/converters/groupConverter.py

```diff
@@ -141,45 +149,177 @@ def build_fhir_member(cls,fhir_family,imis_family):
 
     @classmethod
     def build_fhir_quantity(cls,fhir_family,imis_family):
-        quantity=Insuree.objects.filter(family__uuid=imis_family.uuid).count()
-        fhir_family.quantity=quantity
-        
-    @classmethod
-    def build_fhir_addresses(cls, fhir_family, imis_family):
-        addresses = []
-        if imis_family.address is not None:
-            current_address = cls.build_fhir_address(imis_family.address, "home",
-                                                     "physical")
-            addresses.append(current_address)
-        if type(fhir_family.address) is not list:
-            fhir_family.address = addresses
-        else:
-            fhir_family.address.append(addresses)
+        quantity = Insuree.objects.filter(family__uuid=imis_family.uuid).count()
+        fhir_family.quantity = quantity
 
     @classmethod
-    def build_imis_addresses(cls, imis_family, fhir_family):
-        addresses = fhir_family.address
-        if addresses is not None:
-            for address in addresses:
-                if address.type == "physical":
-                    imis_family.current_address = address.text
-                elif address.type == "both":
-                    imis_family.geolocation = address.text
+    def build_fhir_extentions(cls, fhir_family, imis_family, reference_type):
+        fhir_family.extension = []
+
+        def build_extension(fhir_family, imis_family, value):
+            extension = Extension.construct()
+            if value == "group-address":
+                extension.url = f"{GeneralConfiguration.get_system_base_url()}StructureDefinition/group-address"
+                family_address = cls.build_fhir_address(imis_family.address, "home", "physical")
+                if imis_family.location:
+                    family_address.state = imis_family.location.parent.parent.parent.name
```

Comment:
might be more readable if this will be in  a dedicated fuction

Reply:
ok

Reply:
I will fix this to have this in separate function